include/swift/Sema/ConstraintSystem.h

```diff
@@ -3823,14 +3823,16 @@ class ConstraintSystem {
   /// Given a tuple with a single unlabeled element that represents a pack
   /// expansion (either directly via \c PackExpansionType or through a type
   /// variable constrained to a pack expansion), materialize the pack expansion
-  /// and return its pattern type as a result. The result is a type variable
-  /// because element of the tuple is not required to be resolved at the time of
-  /// the call and operation is delayed until the element is sufficiently
-  /// resolved (see \c simplifyMaterializePackExpansionConstraint)
+  /// and return its pack expansion type as a result. The result is a type
```

Comment:
Promised myself to stop quibbling about this, and here I am 🕯️

"materialize the pack expansion and return its pack expansion type as a result" sounds rather confusing to me. "What does it mean to materialize here?", I’d first ask. Another thing is that the materialization this talks about is a value—not a type—operation in SIL. But this is really just a "is single-element tuple of pack expansion" constraint with a fancy name. How about rephrasing the aforementioned part like so: "return a type variable type representing this element".

Reply:
Doesn’t it return the pattern type of the expansion?

Reply:
Ah, Pavel [is right](https://github.com/apple/swift/blob/f08f86c71617bacbc61f69ce842e284b27036598/lib/Sema/CSSimplify.cpp#L13641-L13643). The constraint kind description needs some attention:

https://github.com/apple/swift/blob/f08f86c71617bacbc61f69ce842e284b27036598/include/swift/Sema/Constraint.h#L221-L223